front/app/containers/Admin/projects/project/nativeSurvey/FormResults/FormResultsQuestion/index.tsx

```diff
@@ -75,44 +68,77 @@ const FormResultsQuestion = ({
           totalSubmissions={totalSubmissions}
           totalResponses={questionResponseCount}
         />
-        {isRanking && <RankingQuestion result={result} />}
-        {isMultipleChoiceAndHasAnswers && (
-          <SurveyBars questionResult={result} colorScheme={COLOR_SCHEME} />
-        )}
-        {hasTextResponses && (
-          <TextQuestion
-            textResponses={textResponses}
-            customFieldId={customFieldId}
-            hasOtherResponses={isMultipleChoiceAndHasAnswers}
-          />
-        )}
-        {hasNumberResponses && (
-          <NumberQuestion numberResponses={numberResponses} />
-        )}
-        {isPointAndHasAnswers && (
-          <PointLocationQuestion
-            pointResponses={pointResponses}
-            mapConfigId={mapConfigId}
-            customFieldId={customFieldId}
-          />
-        )}
-        {isLineAndHasAnswers && (
-          <LineLocationQuestion
-            lineResponses={lineResponses}
-            mapConfigId={mapConfigId}
-            customFieldId={customFieldId}
-          />
-        )}
-        {isPolygonAndHasAnswers && (
-          <PolygonLocationQuestion
-            polygonResponses={polygonResponses}
-            mapConfigId={mapConfigId}
-            customFieldId={customFieldId}
-          />
-        )}
+
+        {(() => {
+          switch (hasAnswersOfType) {
+            case 'ranking':
+              return <RankingQuestion result={result} />;
+            case 'multiselect':
+              return (
+                <SurveyBars
+                  questionResult={result}
+                  colorScheme={COLOR_SCHEME}
+                />
+              );
+            case 'text':
+              if (textResponses) {
+                return (
+                  <TextQuestion
+                    textResponses={textResponses}
+                    customFieldId={customFieldId}
+                    hasOtherResponses={!!answers}
+                  />
+                );
+              }
+              return null;
+            case 'number': {
+              if (numberResponses) {
+                return <NumberQuestion numberResponses={numberResponses} />;
+              }
+              return null;
```

Comment:
Since we have a lot of conditional logic in this file, I wonder if short circuiting wouldn't be clearer. Might just be a preference so I'll leave it to you

```suggestion
              return numberResponses && <NumberQuestion numberResponses={numberResponses} />;
```